web/src/routes/PluginMain.tsx

```diff
@@ -0,0 +1,92 @@
+import {Banner, Flex, FlexItem, Page} from '@patternfly/react-core';
+
+import {Navigate, Outlet, Route, Routes} from 'react-router-dom';
+
+import {NavigationPath} from './NavigationPath';
+import OrganizationsList from './OrganizationsList/OrganizationsList';
+import Organization from './OrganizationsList/Organization/Organization';
+import RepositoryDetails from 'src/routes/RepositoryDetails/RepositoryDetails';
+import RepositoriesList from './RepositoriesList/RepositoriesList';
+import TagDetails from 'src/routes/TagDetails/TagDetails';
+import {useEffect} from 'react';
+import ErrorBoundary from 'src/components/errors/ErrorBoundary';
+import {useQuayConfig} from 'src/hooks/UseQuayConfig';
+import SiteUnavailableError from 'src/components/errors/SiteUnavailableError';
+import NotFound from 'src/components/errors/404';
+import {useCurrentUser} from 'src/hooks/UseCurrentUser';
+import {InfoCircleIcon} from '@patternfly/react-icons';
+
+const NavigationRoutes = [
+  {
+    path: NavigationPath.organizationsList,
+    Component: <OrganizationsList />,
+  },
+  {
+    path: NavigationPath.organizationDetail,
+    Component: <Organization />,
+  },
+  {
+    path: NavigationPath.repositoriesList,
+    Component: <RepositoriesList />,
+  },
+  {
+    path: NavigationPath.repositoryDetail,
+    Component: <RepositoryDetails />,
+  },
+  {
+    path: NavigationPath.tagDetail,
+    Component: <TagDetails />,
+  },
+];
+
+export default function PluginMain() {
+  const quayConfig = useQuayConfig();
+  const {loading, error} = useCurrentUser();
+
+  useEffect(() => {
+    if (quayConfig?.config?.REGISTRY_TITLE) {
+      document.title = quayConfig.config.REGISTRY_TITLE;
+    }
+  }, [quayConfig]);
+
+  if (loading) {
+    return null;
+  }
+  return (
+    <ErrorBoundary hasError={!!error} fallback={<SiteUnavailableError />}>
+      <Page
+        style={{height: '100vh'}}
+      >
+        <Banner variant="info">
+          <Flex
+            spaceItems={{default: 'spaceItemsSm'}}
+            justifyContent={{default: 'justifyContentCenter'}}
+          >
+            <FlexItem>
+              <InfoCircleIcon />
+            </FlexItem>
+            <FlexItem>
+              Please use{' '}
+              <a
+                href="https://forms.gle/M2CtyneF3iaMT5UVA"
+                target="_blank"
+                rel="noreferrer"
+              >
+                this form
+              </a>{' '}
+              to provide feedback on your experience
+            </FlexItem>
+          </Flex>
+        </Banner>
+        <Routes>
+          <Route index element={<Navigate to="/organization" replace />} />
+          {NavigationRoutes.map(({path, Component}, key) => (
+            <Route path={path} key={key} element={Component} />
+          ))}
+          <Route path="*" element={<NotFound />} />
```

Comment:
Just FYI. Routing will have to be eventually adjusted based on the final location of the plugin.